src/System.Security.AccessControl/src/System/Security/AccessControl/ACL.cs

```diff
@@ -323,9 +323,9 @@ private void MarshalHeader( byte[] binaryForm, int offset )
 
             binaryForm[offset + 0] = Revision;
             binaryForm[offset + 1] = 0;
-            binaryForm[offset + 2] = ( byte )( BinaryLength >> 0 );
+            binaryForm[offset + 2] = unchecked(( byte )( BinaryLength >> 0 ));
             binaryForm[offset + 3] = ( byte )( BinaryLength >> 8 );
-            binaryForm[offset + 4] = ( byte )( Count >> 0 );
+            binaryForm[offset + 4] = unchecked(( byte )( Count >> 0 ));
             binaryForm[offset + 5] = ( byte )( Count >> 8 );
```

Comment:
Same here. ACL size and ACE count are unsigned 16-bit integers.
https://msdn.microsoft.com/en-us/library/windows/desktop/aa374931.aspx